docs/architecture/porting-existing-aspnet-apps/understand-update-dependencies.md

```diff
@@ -7,8 +7,37 @@ ms.date: 11/13/2020
 
 # Understand and update dependencies
 
+After identifying the sequence in which the app's individual projects must be migrated, the next step is to understand each project's dependencies and update them if necessary. For platform dependencies, the best way to start is to run the [.NET Portability Analyzer](https://docs.microsoft.com/dotnet/standard/analyzers/portability-analyzer) on the assembly in question, and then look at the detailed results that are generated. You configure to tool to specify one or more target platforms, such as .NET 3.1 or .NET Standard 2.0, and results are provided with details for each platform targeted. Figure 3-4 shows an example of the output from the tool.
+
+![.NET Portability Analyzer report details](./media/Figure3-4.png)
+
+**Figure 3-4. .NET Portability Analyzer report details.**
+
+## Updating Class Library dependencies
+
```

Comment:
You mention platform dependencies, but I think we should also have a section on NuGet dependencies. It's not unusual for customers to need to update their NuGet package references because a package they're using either doesn't support .NET Standard or only supports it with a newer version. There's no good tooling for this, currently, so my recommendation is:

1. Update NuGet packages to `<PackageReference>` format (using VS's converter function) so that they can see just top-level dependencies.
2. Check whether those dependencies support .NET Core or .NET Standard (either in VS or on nuget.org).
3. If necessary, update package versions. This can sometimes introduce breaking changes (especially if the major version of the package changes) which will need to be addressed.
4. In some cases, there are no versions of a given package that works on .NET Core, so they have a couple options:
    1. Keep using the NetFx version. We allow this, but it's risky. They may want to run the Portability Analyzer on the package's binaries to see if they're likely to hit issues and will definitely want to test thoroughly.
    2. Find an alternative package or, if it's open source, consider making the necessary changes to work with netcore/netstandard themeselves.